File: EngageHub/components/eventdetails.py
import reflex as rx
# from ..State.firebaseConfig import r
class RegisterEventData(rx.State):
    form_data: dict = {}

    def handle_submit(self, form_data: dict):
        """Handle the form submit."""
        self.form_data = form_data
       
        print(form_data)
        rx.window_alert("Hello")
        handle_Form_Submit(form_data)
def handle_Form_Submit(form_data):
    from ..State.firebaseConfig import register_for_event
    name = form_data.get("name")  # Access form_data correctly
    email = form_data.get("email")
    pid = form_data.get("pid_number")
    ph_Number = form_data.get("ph_number")
    header = form_data.get("header")
    print("Name:",name)
    print("Email:",email)
    print("PID:",pid)
    print("Phone Number:",ph_Number)
    print("Header:",header)


    try:
            register = register_for_event(header,name,email,pid,ph_Number)
            if register:
                 return rx.chakra.alert(
                 rx.chakra.alert_icon(),
                    rx.chakra.alert_title(
                "Event Registered."
            ),
            status="success",
            ),
            # user = auth.sign_in_with_email_and_password(email, password)
            # UserCredentials.set_my_local_storage = "Vaibhav"
            # add_user_to_firestore("user","user.email")
            # print("User signed in:", user["email"])
            # credentials.username = "Vaibhav"
            # user_state.authenticated = True
            # userEmail = user["email"]
            # print("User Email: ", userEmail)
            # ClStorage.currentUser = "Vaibhav"
            # print("User State: ", ClStorage.currentUser)
            # ClStorage.set_currentUser
            # currentUser("Vaibhav")
            # UserCredentials.add_var()
            # UserStorage.set_user_email("Vaibhav@gmail.com")
            
            # print("Username: ",credentials.username)
            # rx.script("")
            # userCred = auth.current_user()
            return rx.redirect("/")
        # Here you can perform any additional actions after successful sign-in
    except Exception as e:
        print("Error:", e)
        # Handle sign-in error, e.g., display error message to the user
    return rx.text("Sign-in successful!")        
def event_details(event_data) -> rx.Component:
    print("Event Details: ",event_data)
    if event_data is None:
        return rx.text("Loading...")
    
    return rx.container(
        rx.image(src=event_data["url"], width="100%", height="30vw", style={"object-fit": "cover", "border-radius": "10px", "position": "relative"}),
        # rx.link(
        #     rx.button("Register", style={"background-color": "Indigo", "color": "white", "dark":"true", "padding": "15px 30px", "border": "none", "border-radius": "8px", "position": "absolute", "top": "50%", "left": "50%", "transform": "translate(-50%, -50%)", "opacity": "0.8", "font-size": "1.5rem" }),
        #     href=event_data["redirect"],
        #     style={"textDecoration": "none"},
        # ),
        rx.container(
            rx.heading(event_data["header"], style={"color": "black", "font-size": "2rem", "margin-bottom": "0.5rem"}),
            rx.container(
                rx.text(event_data["description"], style={"margin-bottom": "0.2rem", "color": "black"}),
                style={"flex": "1", "margin-right": "2rem"},
            ),
            rx.container(
                rx.text("Date: " + event_data["date"], style={"font-weight": "bold", "color": "black"}),
                rx.text("Time: " + event_data["time"], style={"font-weight": "bold", "color": "black"}),
                rx.text("Location: " + event_data["venue"] + ", " + event_data["location"], style={"font-weight": "bold", "color": "black"}),  # Include location
                style={"margin-top": "0.5rem"},
            ),
            
            rx.vstack(
        rx.form(
            rx.vstack(
                rx.input(
                    placeholder="Name",
                    name="name",
                    # style={"margin-bottom" : "1em"},
                    # margin_bottom="1em",
                    width = "100%"
                ),
                rx.input(
                    placeholder="Enter your email",
                    name="email",
                    type_="email",
                    style={"margin-bottom" : "1em"},
                    margin_bottom="1em",
                    width = "100%"
                ),
                rx.input(name="header", type_="hidden", value=event_data["header"]), 
                # rx.input(
                #     placeholder="Event Number",
                #     name="event_number",
                #     style={"margin-bottom" : "1em"},
                #     margin_bottom="1em",
                #     width = "100%"
                # ),
                rx.input(
                    placeholder="Enter PID",
                    name="pid_number",
                    type_="number",
                    style={"margin-bottom" : "1em"},
                    margin_bottom="1em",
                    width = "100%"
                ),
                rx.input(
                    placeholder="Enter your Phone Number",
                    name="ph_number",
                    type_="number",
                    style={"margin-bottom" : "1em"},
                    margin_bottom="1em",
                    width = "100%"
                ),
                rx.button("Register", type_="submit" , bg = "indigo" , color = "white",variant="outline" , margin_top = "5em" , style={"margin-top" : "3em"}),
            ),
            on_submit=RegisterEventData.handle_submit,
            reset_on_submit=True,
            style={ "padding" : "2em", "width" : "500px", "margin-top" : "3em"}
        ),
    ),
            style={"display": "flex", "flex-direction": "column", "align-items": "center", "gap": "1rem"},
            
        ),
        style={
            "max-width": "100vw",
            "padding": "2rem",
            "display": "flex",
            "flex-direction": "column",
            "align-items": "center",
            "gap": "1rem",
            "background-color": "#ffffff",
            "border": "1px solid #ddd",
            "border-radius": "10px",
            "box-shadow": "0 4px 8px rgba(0, 0, 0, 0.1)",
        },
    )

File: .web/pages/eventdetails.js
/** @jsxImportSource @emotion/react */


import { Fragment } from "react"
import { Fragment_fd0e7cb8f9fb4669a6805377d925fba0 } from "/utils/stateful_components"
import { Box, Breadcrumb, BreadcrumbItem, BreadcrumbLink, Button, Container, Flex, Heading, HStack, Image as ChakraImage, Link, Text } from "@chakra-ui/react"
import NextLink from "next/link"
import "focus-visible/dist/focus-visible"
import { AddIcon } from "@chakra-ui/icons"
import NextHead from "next/head"



export default function Component() {

  return (
    <Fragment>
  <Fragment_fd0e7cb8f9fb4669a6805377d925fba0/>
  <Container sx={{"max-width": "100vw"}}>
  <Box sx={{"backdropFilter": "auto", "backdropBlur": "lg", "p": "4", "position": "sticky", "top": "0", "zIndex": "100"}}>
  <HStack justify={`space-between`}>
  <HStack>
  <Breadcrumb>
  <BreadcrumbItem>
  <BreadcrumbLink as={NextLink} href={`/`}>
  <Heading size={`xl`} sx={{"color": "indigo"}}>
  {`EngageHub`}
</Heading>
</BreadcrumbLink>
</BreadcrumbItem>
</Breadcrumb>
</HStack>
  <HStack>
  <Link as={NextLink} href={`/eventgallery`} sx={{"button": true}}>
  <Button colorScheme={`none`} size={`lg`} sx={{"color": "indigo"}}>
  {`Event Gallery`}
</Button>
</Link>
  <Link as={NextLink} href={`/about`} sx={{"button": true}}>
  <Button colorScheme={`none`} size={`lg`} sx={{"color": "indigo"}}>
  {`About Us`}
</Button>
</Link>
</HStack>
  <HStack spacing={`8`}>
  <Link as={NextLink} href={`/signIn`} sx={{"color": "indigo", "mr": "2"}}>
  {`Login`}
</Link>
  <Link as={NextLink} href={`/signUp`} sx={{"button": true}}>
  <Button sx={{"bg": "indigo", "color": "white"}}>
  {`SignUp`}
</Button>
</Link>
  <Link as={NextLink} href={`/admin`} sx={{"button": true}}>
  <Button sx={{"bg": "indigo", "color": "white"}}>
  {`Admin`}
</Button>
</Link>
</HStack>
</HStack>
</Box>
  <Container sx={{"max-width": "100vw", "padding": "2rem", "display": "flex", "flex-direction": "column", "align-items": "center", "gap": "1rem", "background-color": "#ffffff", "border": "1px solid #ddd", "border-radius": "10px", "box-shadow": "0 4px 8px rgba(0, 0, 0, 0.1)"}}>
  <ChakraImage src={`https://i.ibb.co/SBgYbZt/audience-1853662-640.jpg`} sx={{"object-fit": "cover", "border-radius": "10px", "position": "relative", "width": "100%", "height": "30vw"}}/>
  <Link as={NextLink} href={`/register`} sx={{"textDecoration": "none"}}>
  <Button sx={{"background-color": "Indigo", "color": "white", "dark": "true", "padding": "15px 30px", "border": "none", "border-radius": "8px", "position": "absolute", "top": "50%", "left": "50%", "transform": "translate(-50%, -50%)", "opacity": "0.8", "font-size": "1.5rem"}}>
  {`Register`}
</Button>
</Link>
  <Container sx={{"display": "flex", "flex-direction": "column", "align-items": "center", "gap": "1rem"}}>
  <Heading sx={{"color": "black", "font-size": "2rem", "margin-bottom": "0.5rem"}}>
  {`TantraUtsav`}
</Heading>
  <Container sx={{"flex": "1", "margin-right": "2rem"}}>
  <Text sx={{"margin-bottom": "0.2rem", "color": "black"}}>
  {`EngageHub organized a 3D Modeling Workshop using Blender on 16th February at 5 PM. The workshop taught participants the magic of creating stunning 3D models and animations using Blender. It was suitable for both beginners and experienced users. The event was followed by a blender-render competition, which added to the excitement.`}
</Text>
</Container>
  <Container sx={{"margin-top": "0.5rem"}}>
  <Text sx={{"font-weight": "bold", "color": "black"}}>
  {`Date: February 16, 2023`}
</Text>
  <Text sx={{"font-weight": "bold", "color": "black"}}>
  {`Time: 9 PM`}
</Text>
  <Text sx={{"font-weight": "bold", "color": "black"}}>
  {`Location: Mira , Bhayander`}
</Text>
</Container>
</Container>
</Container>
  <Box sx={{"background-color": "indigo", "text-align": "center", "padding": "1rem"}}>
  <Flex sx={{"margin-bottom": "1rem"}}>
  <AddIcon sx={{"name": "fab fa-twitter", "color": "white", "size": "lg", "mr": "2"}}/>
  <AddIcon sx={{"name": "fab fa-facebook-f", "color": "white", "size": "lg", "mr": "2"}}/>
  <AddIcon sx={{"name": "fab fa-instagram", "color": "white", "size": "lg", "mr": "2"}}/>
  <AddIcon sx={{"name": "fab fa-linkedin-in", "color": "white", "size": "lg", "mr": "2"}}/>
</Flex>
  <Text sx={{"color": "white", "size": "sm", "mb": "2"}}>
  {`Connect with us on social media`}
</Text>
  <Text sx={{"color": "white", "size": "sm", "mb": "2"}}>
  {`For inquiries, contact us at info@engagehub.com`}
</Text>
  <Text sx={{"color": "white", "size": "sm"}}>
  {`© 2024 EngageHub`}
</Text>
</Box>
</Container>
  <NextHead>
  <title>
  {`Reflex App`}
</title>
  <meta content={`A Reflex app.`} name={`description`}/>
  <meta content={`favicon.ico`} property={`og:image`}/>
</NextHead>
</Fragment>
  )
}
